### PR TITLE
Feat/srs 542

### PR DESCRIPTION
**Pull request by jaspalsingh-aot**

### CODE REVIEW

File: backend/sites/src/app/dto/sitePartics.dto.ts [hunk 1]
```diff
@@ -10,9 +13,9 @@ export class SiteParticsResponse extends ResponseDto {
 }
 
 @ObjectType()
-export class SiteParticsDto {
+export class SiteParticsDto extends ChangeAuditObjectTypeDTO {
   @Field()
-  guid: string;
+  partiRoleId: string;
```
Comment (nikhila-aot) [inline backend/sites/src/app/dto/sitePartics.dto.ts]: check the spelling, is it partics?

File: backend/sites/src/app/dto/sitePartics.dto.ts [hunk 2]
```diff
@@ -50,7 +56,7 @@ export class SiteParticsDto {
 @InputType()
 export class SiteParticsInputDto extends ChangeAuditEntityDTO {
   @Field()
-  guid: string;
+  partiRoleId: string;
```
Comment (nikhila-aot) [inline backend/sites/src/app/dto/sitePartics.dto.ts]: should it be partics?

File: frontend/site-registry/src/app/features/details/associates/Associate.tsx [hunk 3]
```diff
@@ -72,62 +94,189 @@ const Associate :React.FC<IComponentProps> = ({showPending = false}) => {
...
-
+  // Handle user type based on username
   useEffect(() => {
-    const userType = loggedInUser?.profile.preferred_username?.includes('bceid')
-      ? UserType.External
-      : loggedInUser?.profile.preferred_username?.includes('idir')
-        ? UserType.Internal
-        : UserType.External;
-    setUserType(userType);
+    if (loggedInUser?.profile.preferred_username?.includes('bceid')) {
```
Comment (midhun-aot) [inline frontend/site-registry/src/app/features/details/associates/Associate.tsx:99]: Please use the helper method in utility.ts

File: backend/sites/src/app/services/site/site.service.ts [hunk 4]
```diff
@@ -549,21 +942,28 @@ export class SiteService {
...
+                    notation.srAction === SRApprovalStatusEnum.PRIVATE
+                      ? UserActionEnum.DEFAULT
+                      : UserActionEnum.UPDATED,
+                  whenUpdated: new Date(),
+                  whoUpdated: userInfo ? userInfo.givenName : '',
+                },
+              });
+            } else {
+              console.log(
+                `There is no event in database againts event id : ${notation.id}`,
```
Comment (nikhila-aot) [inline backend/sites/src/app/services/site/site.service.ts]: use the logger to log messages

File: frontend/site-registry/src/app/features/details/associates/Associate.tsx (hunk 3 above)
Comment (jaspalsingh-aot) [reply frontend/site-registry/src/app/features/details/associates/Associate.tsx:99]: I am waiting for your changes to merge as we discussed regarding use of UserRoleType Enum as we already have UserType Enum. Once you fix that I will change everywhere as UserRoleType is not required.

File: backend/sites/src/app/services/site/site.service.ts [hunk 5]
```diff
@@ -22,8 +22,11 @@ import { HistoryLog } from '../../entities/siteHistoryLog.entity';
 import { LandHistoryService } from '../landHistory/landHistory.service';
```
Comment (nikhila-aot) [inline backend/sites/src/app/services/site/site.service.ts:22]: recheck this file

File: backend/sites/src/app/services/site/site.service.ts [hunk 6]
```diff
@@ -352,28 +360,40 @@ export class SiteService {
...
-                await transactionalEntityManager.save(
-                  SiteAssocs,
+              if (siteAssociations && siteAssociations.length > 0) {
+                await this.processSiteAssociated(
                   siteAssociations,
+                  userInfo,
+                  transactionalEntityManager,
                 );
               } else {
                 console.log('No changes To Site Associations');
```
Comment (nikhila-aot) [inline backend/sites/src/app/services/site/site.service.ts]: replace console.log with logger

**Review by nikhila-aot — CHANGES_REQUESTED** [review]

File: backend/sites/src/app/services/site/site.service.ts (hunk 4 above)
Comment (jaspalsingh-aot) [reply backend/sites/src/app/services/site/site.service.ts]: done

File: backend/sites/src/app/dto/sitePartics.dto.ts (hunk 2 above)
Comment (jaspalsingh-aot) [reply backend/sites/src/app/dto/sitePartics.dto.ts]: done

File: backend/sites/src/app/dto/sitePartics.dto.ts (hunk 1 above)
Comment (jaspalsingh-aot) [reply backend/sites/src/app/dto/sitePartics.dto.ts]: done

File: backend/sites/src/app/services/site/site.service.ts (hunk 5 above)
Comment (jaspalsingh-aot) [reply backend/sites/src/app/services/site/site.service.ts:22]: didn't change anything here

File: backend/sites/src/app/services/site/site.service.ts (hunk 6 above)
Comment (jaspalsingh-aot) [reply backend/sites/src/app/services/site/site.service.ts]: done

**Review by nikhila-aot — APPROVED** [review]

**Pull request merged**